docs/dev_guide/code_standards.rst

```diff
@@ -1,33 +1,64 @@
 
-All code that is part of the SunPy project should follow The Style Guide for
-Python (`PEP 8 <http://www.python.org/dev/peps/pep-0008/>`_) and
-the `coding style and conventions proposed by Astropy
-<http://docs.astropy.org/en/stable/development/codeguide.html#coding-style-conventions>`_.
-Additionally, all
-code that goes in the trunk should be checked using `PyLint
-<http://www.logilab.org/card/pylint_manual>`_. PyLint is an open source tool
-which analyzes Python code and checks for compliance with PEP8, as well as
-common coding errors and other potentially confusing or erroneous code
-statements. Checking the SunPy trunk code this helps to ensure some baseline
-level of quality and consistency for the code, and also helps to prevent
-potential problems from slipping through the cracks into the production code.
-
-If you followed the installation instructions for devs, pylint should already be
-installed on your system. To run PyLint on a file, simply call pylint from the
-command-line, passing in the name of the file you wish to check: ::
+The purpose of the page is to describe the standards that are expected of all code in sunpy.
+All potential developers should read and abide by the following standards.
+Code which does not follow these standards closely will not be accepted.
+
+Language Standard
+-----------------
+
+English is the default language for all documentation strings and inline commands.
+Variables names should also be based on English words.
+In addition, the standard for spelling is American English.
+
+-  Packages must be compatible with Python 2.7, and 3.x (for 3.x compatibility, the six package should be used for new code).
+-  The code *should* be importable with no dependencies other than the Python Standard Library, NumPy, SciPy, matplotlib, and
+   components already required by SunPy.
+   Adding dependencies to SunPy will be considered but are highly discouraged.
+-  Documentation strings must be present for all public classes/methods/functions, and must follow the form outlined in the “Documentation Guidelines” document [tbc].
+   Additionally, examples or tutorials in the package documentation are strongly recommended.
+-  Unit tests are required for all public methods and functions, and should adhere to the standards set in the “Testing Guidelines”
+   document [tbc].
+-  C extensions will be allowed only when they provide a significant performance enhancement over pure python.
+   When C extensions are used, the python interface must meet interface guidelines, and the use of Cython is strongly recommended.
+-  If an external C library is needed, the source code for the library should be bundled with the SunPy core.
+   Additionally, the package must be compatible with using a system-installed library.
+-  Packages can include data in [path tbd] as long as it is less than about 100 kb.
+   If the data exceeds this size, it should be hosted outside the source code repository and downloaded using the approved
```

Comment:
something about sample data and that you can't use sample data for tests only docs